newton/_src/solvers/xpbd_rod/kernels_math.py

```
@@ -0,0 +1,596 @@
+    v1: wp.vec3,
+) -> tuple[wp.vec3, wp.vec3]:
+    """Multiply a 2x2 block matrix by a 2-block vector."""
+    top = _mat33_mul_vec3(A, v0) + _mat33_mul_vec3(B, v1)
+    bot = _mat33_mul_vec3(C, v0) + _mat33_mul_vec3(D, v1)
+    return top, bot
+
+
+@wp.func
+def _block_solve(
+    A: wp.mat33,
+    B: wp.mat33,
```

Ruff (0.15.4)

[warning] Unused function argument: `B`

(ARG001)